shell/browser/atom_browser_context.cc

```diff
@@ -342,6 +344,42 @@ AtomBrowserContext::GetURLLoaderFactory() {
   return url_loader_factory_;
 }
 
+class AuthResponder : public network::mojom::TrustedAuthClient {
+ public:
+  AuthResponder(
+      mojo::PendingReceiver<network::mojom::TrustedAuthClient> auth_client)
+      : receiver_(this, std::move(auth_client)) {}
+  ~AuthResponder() override = default;
+
+ private:
+  void OnAuthRequired(
+      const base::Optional<::base::UnguessableToken>& window_id,
+      uint32_t process_id,
+      uint32_t routing_id,
+      uint32_t request_id,
+      const ::GURL& url,
+      bool first_auth_attempt,
+      const ::net::AuthChallengeInfo& auth_info,
+      ::network::mojom::URLResponseHeadPtr head,
+      mojo::PendingRemote<network::mojom::AuthChallengeResponder>
+          auth_challenge_responder) override {
+    api::URLRequest* url_request = api::URLRequest::FromID(routing_id);
+    if (url_request) {
+      url_request->OnAuthRequired(url, first_auth_attempt, auth_info,
+                                  std::move(head),
+                                  std::move(auth_challenge_responder));
+    }
+  }
+
+  mojo::Receiver<network::mojom::TrustedAuthClient> receiver_;
+};
+
+void AtomBrowserContext::OnLoaderCreated(
+    int32_t request_id,
+    mojo::PendingReceiver<network::mojom::TrustedAuthClient> auth_client) {
+  new AuthResponder(std::move(auth_client));
```

Comment:
Who owns `AuthResponder` ?

Reply:
good question... it should probably be a [strong binding](https://cs.chromium.org/chromium/src/mojo/public/cpp/bindings/strong_binding.h?l=6&rcl=4644ed252018ddf3800d5e652bb7ab376b57c021).

Reply:
by which i mean, [self-owned receiver](https://cs.chromium.org/chromium/src/mojo/public/cpp/bindings/self_owned_receiver.h?l=6&rcl=4644ed252018ddf3800d5e652bb7ab376b57c021)